src/blockencodings.h

```diff
@@ -153,43 +157,21 @@ class CBlockHeaderAndShortTxIDs {
 
     size_t BlockTxCount() const { return shorttxids.size() + prefilledtxn.size(); }
 
-    ADD_SERIALIZE_METHODS;
-
-    template <typename Stream, typename Operation>
-    inline void SerializationOp(Stream& s, Operation ser_action) {
-        READWRITE(header);
-        READWRITE(nonce);
-
-        uint64_t shorttxids_size = (uint64_t)shorttxids.size();
-        READWRITE(COMPACTSIZE(shorttxids_size));
-        if (ser_action.ForRead()) {
-            size_t i = 0;
-            while (shorttxids.size() < shorttxids_size) {
-                shorttxids.resize(std::min((uint64_t)(1000 + shorttxids.size()), shorttxids_size));
-                for (; i < shorttxids.size(); i++) {
-                    uint32_t lsb = 0; uint16_t msb = 0;
-                    READWRITE(lsb);
-                    READWRITE(msb);
-                    shorttxids[i] = (uint64_t(msb) << 32) | uint64_t(lsb);
-                    static_assert(SHORTTXIDS_LENGTH == 6, "shorttxids serialization assumes 6-byte shorttxids");
-                }
-            }
-        } else {
-            for (size_t i = 0; i < shorttxids.size(); i++) {
-                uint32_t lsb = shorttxids[i] & 0xffffffff;
-                uint16_t msb = (shorttxids[i] >> 32) & 0xffff;
-                READWRITE(lsb);
-                READWRITE(msb);
-            }
-        }
-
-        READWRITE(prefilledtxn);
+    template <typename Stream>
+    void Serialize(Stream& s) const
+    {
+        s << header << nonce << Using<VectorFormatter<Uint48Formatter>>(shorttxids) << prefilledtxn;
+    }
 
-        if (BlockTxCount() > std::numeric_limits<uint16_t>::max())
+    template <typename Stream>
+    inline void Unserialize(Stream& s)
+    {
+        static_assert(SHORTTXIDS_LENGTH == 6, "shorttxids serialization assumes 6-byte shorttxids");
+        s >> header >> nonce >> Using<VectorFormatter<Uint48Formatter>>(shorttxids) >> prefilledtxn;
```

Comment:
In commit "Convert blockencodings.h to new serialization framework" (d46dfeccaf1992ba92bff7e0d4988ae3f66b327a)

Maybe a matter of preference, but it might be nice to avoid needing to repeat `s << header...`  `ss >> header...` serialization order. Could do:

```c++
SERIALIZE_METHODS(CBlockHeaderAndShortTxIDs, obj)
{
    static_assert(SHORTTXIDS_LENGTH == 6, "shorttxids serialization assumes 6-byte shorttxids");
    READWRITE(obj.header, obj.nonce, Using<VectorFormatter<Uint48Formatter>>(obj.shorttxids), obj.prefilledtxn);
    if (ser_action.ForRead()) {
        if (obj.BlockTxCount() > std::numeric_limits<uint16_t>::max()) {
            throw std::ios_base::failure("indexes overflowed 16 bits");
        }
        obj.FillShortTxIDSelector();
    }
}
```

Reply:
Ah indeed, that works because `shorttxidkX` are mutable, which they probably shouldn't be. But no need to fix that here.

Done.